### PR TITLE
KAFKA-15746: KRaft support in ControllerMutationQuotaTest

### PR DESCRIPTION
**Pull request by linzihao1999**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/ControllerMutationQuotaTest.scala [hunk 1]
```diff
@@ -139,20 +143,23 @@ class ControllerMutationQuotaTest extends BaseRequestTest {
       val (_, errors) = createTopics(Map("topic" -> 1), StrictDeleteTopicsRequestVersion)
       assertEquals(Set(Errors.NONE), errors.values.toSet)
 
-      // Metric must be there with the correct config
-      waitQuotaMetric(principal.getName, ControllerMutationRate)
+      if (!isKRaftTest()) {
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/server/ControllerMutationQuotaTest.scala:146]: Why are we only doing these checks in ZK mode? Isn't the controller mutation quota also working in KRaft mode?

**Review by mimaison — COMMENTED** [review]
Thanks for the PR, I left a question.

**Comment by mimaison** [comment]
@linzihao1999 Are you planning to address the review comments? Let us know if you have issues. Thanks

**Comment by mimaison** [comment]
@linzihao1999 Can you follow up this PR?

**Comment by linzihao1999** [comment]
@mimaison Sorry for stale this issue, i can follow it. If your has another arrangement, fell free to take over.

**Comment by mimaison** [comment]
@linzihao1999 No worries. It's not really urgent but it would be nice to get it done in the next month or so.

**Comment by mimaison** [comment]
Done in https://github.com/apache/kafka/pull/16620. Closing

**Pull request closed without merging**